### PR TITLE
Standard lib not fully tested

### PR DESCRIPTION
**Pull request by BooTheGhoster**

### CODE REVIEW

File: include/stdlib.fg [hunk 1]
```diff
@@ -1,66 +1,191 @@
...
+number _nozzleDiameter = 0;
+number _layerHeight = 0;
+point _currentPosition = [0,0,0];
+// Default for ultimaker 2
+//number _filamentDiameter = 2.85;
+//number _nozzleDiameter = 0.4;
+//number _layerHeight = 0.1;
+//point _currentPosition = [0,0,0];
+
+number _calculateFilamentVolumeNeeded(number length){
```
Comment (mads256h) [inline include/stdlib.fg:13]: Is `_distanceToFilamentLength` a more descriptive name?

File: include/stdlib.fg [hunk 2]
```diff
@@ -1,66 +1,191 @@
...
+    number filamentNeeded = length * _nozzleDiameter * _layerHeight;
+    number filamentArea = (_filamentDiameter / 2)^2 * pi;
+    if (!(filamentArea == 0)){
+        number filamentMM = filamentNeeded / filamentArea;
+        return filamentMM;
+    }
+    else{
+        return 0;
```
Comment (mads256h) [inline include/stdlib.fg:21]:
```suggestion
    if (filamentArea != 0) {
        return filamentNeeded / filamentArea;
    } else {
        return 0;
```

File: include/stdlib.fg [hunk 3]
```diff
@@ -1,66 +1,191 @@
...
+    gcodeLine(endpoint, filamentMm);
+    _currentPosition = endpoint;
+    return _currentPosition;
 };
 
 point _arc(point center, point wantedEndpoint, bool is_cw){
-    number i = center.x - currentPosition.x;
-    number j = center.y - currentPosition.y;
+    number i = center.x - _currentPosition.x;
+    number j = center.y - _currentPosition.y;
```
Comment (mads256h) [inline include/stdlib.fg:39]: This is a bug
```suggestion
    number i = center.x + _currentPosition.x;
    number j = center.y + _currentPosition.y;
```

or even better
```suggestion
point abs_center = center + currentPosition;
```

File: include/stdlib.fg [hunk 4]
```diff
@@ -1,66 +1,191 @@
...
+    // We calculate the point where the circle ends
     point vector = wantedEndpoint - center;
     point normalizedVector = [vector.x / sqrt(vector.x^2 + vector.y^2),vector.y / sqrt(vector.x^2 + vector.y^2),0];
```
Comment (mads256h) [inline include/stdlib.fg:42]:
```suggestion
    point normalizedVector = vector / length(vector);
```
or even better create a normalized function in math.fg

File: include/stdlib.fg [hunk 5]
```diff
@@ -1,66 +1,191 @@
...
     point realEndpoint = (normalizedVector * radius) + center;
+    // We define the Gcode helper functions
+
+    gcode gcodeCArc(point endpoint, number i, number j, number e){
+        G2 E[e] X[endpoint.x] Y[endpoint.y] I[i] J[j]
+    };
+
+    gcode gcodeCWArc(point endpoint, number i, number j, number e){
+        G3 E[e] X[endpoint.x] Y[endpoint.y] I[i] J[j]
+    };
```
Comment (mads256h) [inline include/stdlib.fg:53]: Move this down to just before they are used

File: include/stdlib.fg [hunk 6]
```diff
@@ -1,66 +1,191 @@
...
+
+    // We calculate the other sides and the C angle (in radians) with trigonometry
+    number c = sqrt((realEndpoint.x - _currentPosition.x)^2 + (realEndpoint.y - _currentPosition.y)^2);
```
Comment (mads256h) [inline include/stdlib.fg:56]:
```suggestion
    number c = distance2D(realEndpoint, _currentPosition);
```

File: include/stdlib.fg [hunk 7]
```diff
@@ -1,66 +1,191 @@
...
+    number cosC = (2 *radius^2 - c^2) / (2 * radius^2);
+    number C = arccos(cosC);
+    //We calculate the circumference and multiply it by the percentage of the circle that is filled out
+    number circumference = radius * 2 * pi;
+
+    //We now know the filament needed for each of the filled edges and now we need to calculate whether clockwise or
+    //counter clockwise has the biggest side
```
Comment (mads256h) [inline include/stdlib.fg:63]:
```suggestion
    // Calculate the circumference and multiply it by the percentage of the circle that is filled out
    number circumference = radius * 2 * pi;

    // We now know the filament needed for each of the filled edges and now we need to calculate whether clockwise or
    // counter clockwise has the biggest side
```

File: include/stdlib.fg [hunk 8]
```diff
@@ -1,66 +1,191 @@
...
+    number smallestFilamentMM = _calculateFilamentVolumeNeeded(smallestFilledCircumference);
 
+    // We figure out the function between the current position and center of circle
+    number a = 0;
+    number b = 0;
+
+    bool clockwiseIsSmallest = false;
+    bool cClockwiseIsSmallest = false;
+
+    //If current position.x == center.x both lines will be the same length
```
Comment (mads256h) [inline include/stdlib.fg:76]:
```suggestion
    // If current position.x == center.x both lines will be the same length
```

File: include/stdlib.fg [hunk 9]
```diff
@@ -1,66 +1,191 @@
...
         };
-        gcodeCWArc(realEndpoint, i, j);
     };
-    currentPosition = realEndpoint;
-    return currentPosition;
+    _currentPosition = realEndpoint;
+    return _currentPosition;
+};
+
+point arc(point center, point wantedEndpoint){
```
Comment (mads256h) [inline include/stdlib.fg:108]: I want this to be named so you know you do a clockwise arc

File: include/stdlib.fg [hunk 10]
```diff
@@ -1,66 +1,191 @@
...
+    return _arc(center, wantedEndpoint, true);
+};
+
+point cArc(point center, point wantedEndpoint){
```
Comment (mads256h) [inline include/stdlib.fg:112]: The same for this one but for counter-clockwise

**Review by mads256h — DISMISSED** [review]

File: include/stdlib.fg (hunk 3 above)
Comment (BooTheGhoster) [reply include/stdlib.fg:39]: Det er korrekt som det er nu

**Review by Malgreen — APPROVED** [review]
LGTM+

**Pull request merged**